src/Storage/Storage.Management/Models/PSBlobServiceProperties.cs

```diff
@@ -292,15 +295,37 @@ public CorsRule ParseCorsRule()
         /// <returns>String Array</returns>
         private static string[] ListToArray(IList<string> stringList)
         {
+            string[] stringArray = new string[stringList.Count];
             if (null == stringList)
             {
-                return null;
+                return stringArray;
             }
 
-            string[] stringArray = new string[stringList.Count];
             stringList.CopyTo(stringArray, 0);
             return stringArray;
         }
+
+        private List<string> ArrayToList(string[] stringArray)
+        {
+            return stringArray == null ? new List<string>() : new List<string>(stringArray);
+        }
+
+        private List<string> ParseAllowedMethods(string[] allowedMethods)
+        {
+            List<string> methods = new List<string>();
+            foreach(string method in allowedMethods)
+            {
+                if (AllowedMethodList.Contains(method.ToUpper())) {
+                    methods.Add(method.ToUpper());
+                } 
+                else
+                {
+                    throw new InvalidOperationException(string.Format("{0} is an invalid HTTP method.", method));
```

Comment:
I am doubt should we return error here. 
It might block the API when some new method added in the future.
When the method not exist, we might can just add the orignal value.

Reply:
I've added the check here just to make sure we can throw the exception at client side just as what dataplane cmdlets are doing. But it's true that whenever a new value is added in the future, the value will be added to the list here too.
Regarding "When the method not exist, we might can just add the original value.", I think in this case we just don't need to check if the method exists, but add the value to the list anyways

Reply:
Thus I guess the decision here should be - do we want to align with the dataplane cmdlets to check for it on the client side? Or do we just want the server to throw the exception?